go/vt/mysqlctl/tmutils/schema.go

```diff
@@ -63,94 +63,130 @@ func (tds TableDefinitions) Swap(i, j int) {
 	tds[i], tds[j] = tds[j], tds[i]
 }
 
-// FilterTables returns a copy which includes only whitelisted tables
+// TableFilter is a filter for table names and types.
+type TableFilter struct {
+	includeViews bool
+
+	filterTables bool
+	tableNames   []string
+	tableREs     []*regexp.Regexp
+
+	filterExcludeTables bool
+	excludeTableNames   []string
+	excludeTableREs     []*regexp.Regexp
+}
+
+// NewTableFilter creates a TableFilter for whitelisted tables
 // (tables), no blacklisted tables (excludeTables) and optionally
 // views (includeViews).
-func FilterTables(sd *tabletmanagerdatapb.SchemaDefinition, tables, excludeTables []string, includeViews bool) (*tabletmanagerdatapb.SchemaDefinition, error) {
-	copy := *sd
-	copy.TableDefinitions = make([]*tabletmanagerdatapb.TableDefinition, 0, len(sd.TableDefinitions))
+func NewTableFilter(tables, excludeTables []string, includeViews bool) (*TableFilter, error) {
+	f := &TableFilter{
+		includeViews: includeViews,
+	}
 
 	// Build a list of regexp to match table names against.
 	// We only use regexps if the name starts and ends with '/'.
 	// Otherwise the entry in the arrays is nil, and we use the original
 	// table name.
-	var tableRegexps []*regexp.Regexp
 	if len(tables) > 0 {
-		tableRegexps = make([]*regexp.Regexp, len(tables))
-		for i, table := range tables {
-			if strings.HasPrefix(table, "/") {
+		f.filterTables = true
+		for _, table := range tables {
+			if strings.HasPrefix(table, "/") && strings.HasSuffix(table, "/") {
```

Comment:
Did we add this extra check for a specific reason.
It is a breaking change: local example breaks because we are using /.* After removing this check local example passed on my system.
Iirc we have other places in the code where we use /.* without the trailing / , though I don't know if they use this function.

Reply:
Since this was already merged, hotfixed just the trailing regexp change: https://github.com/vitessio/vitess/pull/6259/

Reply:
@rohit-nayak-ps 
yes, the comment just above specifically says what the behavior should be. it just never was tested.

`// We only use regexps if the name starts and ends with '/'.`

Reply:
I did read the comment and understand what that code is doing.
My question was: do you require trailing slashes in your new code and would removing this check break any of your PRs? I wanted to be aware if my hotfix would break anything.